types/module/module.go

```diff
@@ -174,6 +174,12 @@ type AppModule interface {
 	// RegisterServices allows a module to register services
 	RegisterServices(Configurator)
 
+	// ConsensusVersion tracks state-breaking versions of the module
+	ConsensusVersion() uint64
+
+	// MigrateStore performs in-place store migrations.
+	MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, fromVersion uint64) error
```

Comment:
Instead, how about add a method to `Configurator`: `RegisterMigration(version uint64, handler func(ctx sdk.Context) error)`. Then the module will have the `StoreKey` and it should be possible to do that `ModuleManager` thing more cleanly.

Reply:
ok, I did that in my last commit, ModuleManager looks cleaner. Note that the `Configurator` is not just a "configurator" anymore, since it also holds the logic to actually run the migrations. I don't think we can move that `RunMigrations` function elsewhere unless we make its storeKeys public somehow.